### PR TITLE
DEV: Add `skip_migrations` param when importing remote theme

### PR DESCRIPTION
**Pull request by tgxworld**
Why this change?

Importing theme with the `bundle` params is used mainly by
`discourse_theme` CLI in the development environment. However, we do not
want migrations to automatically run in the development environment
and instead want the developer to be intentional about running theme
migrations. As such, this commit adds support for a
`skip_migrations` param when importing a theme with the `bundle` params.

This commit also adds a `migrated` attribute for migrations theme fields
to indicate whether a migrations theme field has been migrated or not.

### CODE REVIEW

File: app/models/theme.rb [hunk 1]
```diff
@@ -874,7 +877,8 @@ def migrate_settings(start_transaction: true)
         record.calculate_diff(res[:settings_before], res[:settings_after])
         record.save!
       end
-      self.save!
+
+      self.reload
```
Comment (OsamaSayegh) [inline app/models/theme.rb:881]: I'm curious, why did you change this to `reload`?

Comment (tgxworld) [reply app/models/theme.rb:881]: O because there is no need to save here and saving was causing the callbacks to fire and ended up publishing redundant MessageBus messages which was causing a test to fail.

Comment (OsamaSayegh) [reply app/models/theme.rb:881]: Ah, I see. Thanks for the explanation.

**Review by OsamaSayegh — APPROVED** [review]

**Comment by tgxworld** [comment]
Thank you @OsamaSayegh :+1:
